### PR TITLE
WindowServer: Scaled backgrounds now supported

### PR DESCRIPTION
**Pull request by christopherdumas**
Completing the code from my earlier configuration PR, I finished getting scaled backgrounds to work properly. As it turns out, `draw_scaled_bitmap` does not work properly for doing dirty_rect backgrounds because it doesn't do clipping, and seems to ignore doing positioning like `blit` operations do, so I implemented a proper `blit_scaled` function. In a subsequent PR I will unify all the blit functions into one, and have them take a `enum` mode.

2019-06-04-173039_1084x1083_scrot

### CODE REVIEW

**Comment by christopherdumas** [comment]
Yeah that was a bad move on my part. Sorry. I'll try to figure out how to fix that.

**Comment by christopherdumas** [comment]
Okay, I fixed that.

File: SharedGraphics/Painter.cpp [hunk 1]
```diff
@@ -204,6 +204,36 @@ void Painter::draw_bitmap(const Point& p, const GlyphBitmap& bitmap, Color color
     }
 }
 
+void Painter::blit_scaled(const Point& position, const GraphicsBitmap& source, const Rect& src_rect, float hscale, float vscale)
```
Comment (awesomekling) [inline SharedGraphics/Painter.cpp]: This API is super awkward. It should instead take the `dst_rect` as an input parameter, and compute the `hscale` and `vscale` internally:
```cpp
void Painter::blit_scaled(const Rect& dst_rect, const GraphicsBitmap& source, const Rect& src_rect)
```
The rest of the code doesn't have to change, just the API.

**Comment by christopherdumas** [comment]
Okay, I fixed that.

File: Base/home/anon/WindowManager.ini [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [Screen]
 Width=1080
-Height=768
+Height=1080
```
Comment (awesomekling) [inline Base/home/anon/WindowManager.ini]: Let's leave the default screen size at 1024x768 :)

Comment (christopherdumas) [reply Base/home/anon/WindowManager.ini]: Oh DARN. That wasn't supposed to make it through.

Comment (christopherdumas) [reply Base/home/anon/WindowManager.ini]: Fixed.

**Comment by awesomekling** [comment]
Coolio :)

**Pull request merged**